### PR TITLE
16 packages from framagit.org/zoggy/stog/-/archive/1.0.0/stog-1.0.0.tar.gz

### PR DESCRIPTION
**Pull request by zoggy**
This pull-request concerns:
- `stog.1.0.0`: Static web site compiler, able to handle blog posts as well as
  regular pages or any XML document in general
- `stog_all.1.0.0`: Virtual package to install all Stog libraries, tools and
  plugins
- `stog_asy.1.0.0`: Stog plugin to include Asymptote results in documents
- `stog_dot.1.0.0`: Stog plugin to generate and include graphviz graphs in
  documents
- `stog_extern.1.0.0`: Stog plugin to pipe documents in external commands
- `stog_markdown.1.0.0`: Stog plugin to use markdown syntax
- `stog_multi_doc.1.0.0`: Stog plugin to define various documents in one file
- `stog_nocaml.1.0.0`: Stog plugin to block commands executing ocaml code
- `stog_noexec.1.0.0`: Stog plugin to prevent running command with <exec>
- `stog_plugins.1.0.0`: Virtual package to install all Stog plugins
- `stog_rdf.1.0.0`: Stog plugin to generate rdf triples and execute sparql
  queries
- `stog_rel_href.1.0.0`: Stog plugin to generate relative urls
- `stog_server.1.0.0`: Stog server library
- `stog_server_multi.1.0.0`: Stog multi server library
- `stog_sitemap.1.0.0`: Stog plugin to generate a sitemap file
- `stog_writing.1.0.0`: Stog plugin to generate table of contents and
  bibliographies

---
* Homepage: https://www.good-eris.net/stog/
* Source repo: git+https://framagit.org/zoggy/stog.git
* Bug tracker: https://framagit.org/zoggy/stog/issues

---
:camel: Pull-request generated by opam-publish v2.3.0

### CODE REVIEW

**Comment by zoggy** [comment]
"All checks have failed" is a bit exaggerated...
Regarding the bad checksum for iri 0.6.0 (which is very old, dependencies require 1.0.0), I'll have a look (probably same story as in https://github.com/ocaml/opam-repository/issues/25344 .

**Comment by mseri** [comment]
As in that case the old sources are still available at https://opam.robur.coop/cache/md5/40/40574cd88a7fc984adae1839fedb2b10

**Comment by mseri** [comment]
Is this something that requires an upper bound?
```
=== ERROR while compiling stog-writing.0.19.0 ================================#
# context              2.2.0~beta2~dev | linux/x86_64 | ocaml-base-compiler.5.1.1 | file:///home/opam/opam-repository
# path                 ~/.opam/5.1/.opam-switch/build/stog-writing.0.19.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build make all
# exit-code            2
# env-file             ~/.opam/log/stog-writing-7-b389c6.env
# output-file          ~/.opam/log/stog-writing-7-b389c6.out
### output ###
# ocamlfind ocamlc -I +ocamldoc -g -annot -package menhirLib,xtmpl,stog,ocf -rectypes -c bibtex.mli
# ocamlfind: [WARNING] Package `uunf.string': Deprecated. Use the uunf library, the Uunf_string module has moved there.
# menhir bib_parser.mly
# Warning: 5 end-of-stream conflicts were arbitrarily resolved.
# File "<standard.mly>", line 109, characters 18-18:
# Warning: production option(EOF) -> is never reduced.
# Warning: in total, 1 production is never reduced.
# Error: the code back-end requires the type of every nonterminal symbol to be
# known. Please specify the type of every symbol via %type declarations, or
# enable type inference (look up --infer in the manual).
# Type inference is automatically enabled when Menhir is used via Dune,
# provided the dune-project file says (using menhir 2.0) or later.
# The types of the following nonterminal symbols are unknown:
# entry
# field
# list(entry)
# list(field)
# option(COMMA)
# option(EOF)
# make: *** [Makefile:83: bib_parser.mli] Error 1
```

File: packages/stog/stog.1.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    "-j"
+    jobs
+    "--promote-install-files=false"
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+  ["dune" "install" "-p" name "--create-install-files" name]
+]
+dev-repo: "git+https://framagit.org/zoggy/stog.git"
```
Comment (mseri) [inline packages/stog/stog.1.0.0/opam:47]:
```suggestion
conflicts: [ "ocaml-option-bytecode-only" ]
dev-repo: "git+https://framagit.org/zoggy/stog.git"
```
Due to
```

#=== ERROR while compiling stog.1.0.0 =========================================#
# context              2.2.0~beta2~dev | linux/arm32 | ocaml-base-compiler.5.1.1 | pinned(https://framagit.org/zoggy/stog/-/archive/1.0.0/stog-1.0.0.tar.gz)
# path                 ~/.opam/5.1/.opam-switch/build/stog.1.0.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p stog -j 79 --promote-install-files=false @install
# exit-code            1
# env-file             ~/.opam/log/stog-7-63af57.env
# output-file          ~/.opam/log/stog-7-63af57.out
### output ###
# File "exe/dune", line 4, characters 14-20:
# 4 |   (modes byte native)
#                   ^^^^^^
# Error: this mode isn't available
```

**Comment by zoggy** [comment]
For stog_writing 0.19.0, indeed we could add upper bound for menhir version: [20210310]

**Comment by mseri** [comment]
I am assuming that the name changes were intentional:
```
2024-03-14 09:21.02: Warning in stog_writing.1.0.0: Possible name collision with package 'stog-writing'
2024-03-14 09:21.02: Warning in stog_rdf.1.0.0: Possible name collision with package 'stog-rdf'
2024-03-14 09:21.02: Warning in stog-writing.0.19.0: Possible name collision with package 'stog_writing'
```

**Comment by zoggy** [comment]
Yes, this plugin is now part of stog repo and all package names were normalized to use `_` instead of `-`.

**Comment by zoggy** [comment]
I think it's ok, now. Thanks !

**Comment by mseri** [comment]
Thanks!

**Pull request merged**